plugins/woocommerce-blocks/tests/e2e/tests/filter-blocks/rating-filter.block_theme.spec.ts

```diff
@@ -24,6 +24,72 @@ const test = base.extend< {
 
 test.describe( 'Product Filter: Rating Filter Block', () => {
 	test.describe( 'frontend', () => {
+		test( 'clear button is not shown on initial page load', async ( {
+			page,
+			defaultBlockPost,
+		} ) => {
+			await page.goto( defaultBlockPost.link );
+
+			const button = page.locator(
+				'.wp-block-woocommerce-product-filter-clear-button'
+			);
+
+			await expect( button ).toBeHidden();
+		} );
+
+		test( 'clear button appears after a filter is applied', async ( {
+			page,
+			defaultBlockPost,
+		} ) => {
+			await page.goto( `${ defaultBlockPost.link }?rating_filter=1` );
+
+			const button = page.locator(
+				'.wp-block-woocommerce-product-filter-clear-button'
+			);
+
+			await expect( button ).toBeVisible();
+		} );
+
+		test( 'clear button hides after deselecting all filters', async ( {
+			page,
+			defaultBlockPost,
+		} ) => {
+			await page.goto( `${ defaultBlockPost.link }?rating_filter=1` );
+
+			const ratingCheckboxes = page.getByLabel(
+				/Checkbox: Rated \d out of 5/
+			);
+
+			await ratingCheckboxes.nth( 0 ).uncheck();
+
+			const button = page.locator(
+				'.wp-block-woocommerce-product-filter-clear-button'
+			);
+
+			await expect( button ).toBeHidden();
+		} );
+
+		test( 'filters are cleared after clear button is clicked', async ( {
+			page,
+			defaultBlockPost,
+		} ) => {
+			await page.goto( `${ defaultBlockPost.link }?rating_filter=1` );
+
+			const button = page
+				.locator( '.wp-block-woocommerce-product-filter-clear-button' )
+				.locator( 'button' );
+
+			await button.click();
+
+			await new Promise( ( resolve ) => setTimeout( resolve, 1000 ) );
```

Comment:
I would avoid using `setTimeout` to wait for something to occur, because it will most likely cause flakyness. Playwright includes auto-retrying assertions that remove flakiness by waiting until the condition is met ([reference](https://playwright.dev/docs/actionability)).

Reply:
Do you have suggestions around this? Currently without it, it will fail.

Reply:
I am able to get around not using timeouts by explicitly asserting it. However for [this here](https://github.com/woocommerce/woocommerce/pull/47101/files#diff-524ce0ba17c55755ebb47ef00ad631b056f0739fcdfa1c659f61d6ddfa74c0a2R136) I am not sure what's the best way to say "wait for" `data-wc-context` to be updated/changed.

Reply:
Ok I got around this by checking the URL link. Because after clicking "clear", the URL is expected to have no arguments..